src/Huntress/Plugin/Stonks.php

```diff
@@ -0,0 +1,155 @@
+            if (!isset($this->cache[$symbol]) || $this->cache[$symbol]['time'] + $this->ttl < $now) {
+                $get[] = $symbol;
+            }
+        }
+        // Get new data.
+        if ($get) {
+            $data = $this->client->getQuotes($get);
+            foreach ($data as $i => $quote) {
+                // Do not cache errors.
+                if ($quote instanceof Quote) {
+                    $this->cache[$get[$i]] = [
+                        'time'  => $now,
+                        'quote' => $quote,
+                    ];
+                }
+            }
+        }
+        // Format data.
+        $errors = [];
+        $promise = \React\Promise\resolve();
+        foreach ($arg as $symbol) {
+            if (isset($this->cache[$symbol])) {
+                /** @var \Scheb\YahooFinanceApi\Results\Quote $quote */
+                ['time' => $time, 'quote' => $quote] = $this->cache[$symbol];
+                $currency = $quote->getCurrency();
+                $embed = new MessageEmbed();
+                $embed->setTitle("{$symbol} ({$quote->getFullExchangeName()})");
+                $embed->addField(
+                    'Ask / Bid',
+                    "{$currency} {$quote->getAsk()} / {$currency} {$quote->getBid()}"
+                );
+                $embed->addField(
+                    'Market state',
+                    $quote->getMarketState()
+                );
+                $embed->addField(
+                    'Opening price',
+                    sprintf('%s %.2f', $currency, $quote->getRegularMarketOpen())
+                );
+                $embed->addField(
+                    $quote->getMarketState() === 'REGULAR' ? 'Market price' : 'Closing price',
+                    sprintf(
+                        '%s %.2f, (%+.3f%%)',
+                        $currency, $quote->getRegularMarketPrice(),
+                        $quote->getRegularMarketChangePercent()
+                    )
+                );
+                if ($quote->getPostMarketPrice()) {
+                    $embed->addField(
+                        'Post-Market price',
+                        sprintf(
+                            '%s %.2f, (%+.3f%%)',
+                            $currency,
+                            $quote->getPostMarketPrice(),
+                            $quote->getPostMarketChangePercent()
+                        )
+                    );
+                }
+                if ($quote->getPreMarketPrice()) {
+                    $embed->addField(
+                        'Pre-Market price',
+                        sprintf(
+                            '%s %.2f, (%+.3f%%)',
+                            $currency,
+                            $quote->getPreMarketPrice(),
+                            $quote->getPreMarketChangePercent()
+                        )
+                    );
+                }
+                $embed->setFooter(sprintf("Updated: %s", date('Y-m-d H:i:s', $time)));
```

Comment:
this should use discord's built-in embed timestamp functionality if possible, otherwise use `Localization::fetchTimezone()` and Carbon to timezone it appropriately. [Example](https://github.com/sylae/huntress/blob/7b0c6cbdaa01cf1cf9b1a87e1a80d8b258440631/src/Huntress/Plugin/Event.php#L230-L243) in the wild

Reply:
✔️